src/Traits/FactoryTrait.php

```diff
@@ -249,6 +249,58 @@ public static function createFromFormat($format, $time, $tz = null): ChronosInte
         return $dt;
     }
 
+    /**
+     * Creates a ChronosInterface instance from an array of date and time values.
+     *
+     * Allowed values:
+     *  - year
+     *  - month
+     *  - day
+     *  - hour
+     *  - minute
+     *  - second
+     *  - microsecond
+     *  - meridian ('am or 'pm')
+     *  - timezone
+     *
+     * @param (int|string)[] $values Array of date and time values.
+     * @return static
+     */
+    public static function createFromArray($values): ChronosInterface
+    {
+        $values += ['hour' => 0, 'minute' => 0, 'second' => 0, 'microsecond' => 0];
+
+        $formatted = '';
+        if (
+            isset($values['year'], $values['month'], $values['day']) &&
+            (
+                is_numeric($values['year']) &&
+                is_numeric($values['month']) &&
+                is_numeric($values['day'])
+            )
+        ) {
+            $formatted .= sprintf('%d-%02d-%02d', $values['year'], $values['month'], $values['day']);
```

Comment:
Should year be forced to 4 digits to support pre 1000 years?

Reply:
Definitely.